src/sentry/api/endpoints/release_deploys.py

```diff
@@ -0,0 +1,168 @@
+    resources = ListField(required=False)
+
+
+class ReleaseDeploysEndpoint(OrganizationReleasesBaseEndpoint):
+    doc_section = DocSection.RELEASES
+
+    def get(self, request, organization, version):
+        """
+        List a Release's Deploys
+        ````````````````````````
+        Return a list of deploys for a given release.
+        :pparam string organization_slug: the organization short name
+        :pparam string version: the version identifier of the release.
+
+        """
+        try:
+            release = Release.objects.get(
+                version=version,
+                organization=organization
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        allowed_projects = set(
+            self.get_allowed_projects(
+                request,
+                organization
+            ).values_list('id', flat=True)
+        )
+
+        # make sure user has access to at least one project
+        # in release
+        if not [p for p in release.projects.values_list('id', flat=True) if p in allowed_projects]:
+            raise PermissionDenied
+
+        queryset = Deploy.objects.filter(
+            organization_id=organization.id,
+            release=release
+        ).extra(select={
+            'sort': 'COALESCE(date_finished, date_started)',
+        })
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by='-sort',
+            paginator_cls=OffsetPaginator,
+            on_results=lambda x: serialize(x, request.user),
+        )
+
+    def post(self, request, organization, version):
+        """
+        List a Release's Deploys
+        ````````````````````````
+        Return a list of deploys for a given release.
+        :pparam string organization_slug: the organization short name
+        :pparam string version: the version identifier of the release.
+
+        """
+        try:
+            release = Release.objects.get(
+                version=version,
+                organization=organization
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        allowed_projects = set(
+            self.get_allowed_projects(
+                request,
+                organization
+            ).values_list('id', flat=True)
+        )
+
+        allowed_projects_in_release = {
+            p for p in release.projects.values_list('id', flat=True) if p in allowed_projects
+        }
+
+        # make sure user has access to at least one project
+        # in release
+        if not allowed_projects_in_release:
+            raise PermissionDenied
+
+        serializer = DeploySerializer(data=request.DATA)
+
+        if serializer.is_valid():
+            result = serializer.object
+            # TODO(jess) umm should environment be unique across org?
+            # also should we be creating environments if they don't exist?
```

Comment:
@dcramer and/or @mattrobenolt could use your thoughts here. should environment names be unique for an org? right now they are just unique to projects, but that could cause weird issues where we don't know which environment they're deploying to in this endpoint